internal/config/config.go

```diff
@@ -16,54 +16,36 @@ import (
 
 const Filename = ".ldcli-config.yml"
 
-// ConfigFile represents the data stored in the config file.
-type ConfigFile struct {
+type ReadFile func(name string) ([]byte, error)
+
+// Config represents the data stored in the config file.
+type Config struct {
 	AccessToken     string `json:"access-token,omitempty" yaml:"access-token,omitempty"`
 	AnalyticsOptOut *bool  `json:"analytics-opt-out,omitempty" yaml:"analytics-opt-out,omitempty"`
 	BaseURI         string `json:"base-uri,omitempty" yaml:"base-uri,omitempty"`
-	Flag            string `json:"flag,omitempty" yaml:"flag,omitempty"`
 	Environment     string `json:"environment,omitempty" yaml:"environment,omitempty"`
+	Flag            string `json:"flag,omitempty" yaml:"flag,omitempty"`
 	Output          string `json:"output,omitempty" yaml:"output,omitempty"`
 	Project         string `json:"project,omitempty" yaml:"project,omitempty"`
 }
 
-type Config struct {
-	RawConfig map[string]interface{}
-	filename  string
-}
-
 func New(filename string, readFile ReadFile) (Config, error) {
 	data, err := readFile(filename)
 	if err != nil {
 		return Config{}, err
 	}
 
-	rawConfig := make(map[string]interface{}, 0)
-	err = yaml.Unmarshal([]byte(data), &rawConfig)
+	var c Config
```

Comment:
Instead of converting to this generic type, there's a switch statement for updating all valid fields later.